src/newcharacter.cpp

```diff
@@ -1503,8 +1531,58 @@ int set_scenario(WINDOW *w, player *u, int &points)
     ctxt.register_action("NEXT_TAB");
     ctxt.register_action("SORT");
     ctxt.register_action("HELP_KEYBINDINGS");
+    ctxt.register_action("FILTER");
+
+    bool recalc_scens = true;
+    int scens_length;
+    std::string filterstring = "";
+    std::vector<const scenario *> sorted_scens;
 
     do {
+        if (recalc_scens) {
+            sorted_scens.clear();
+            for (scenmap::const_iterator iter = scenario::begin(); iter != scenario::end(); ++iter) {
+                if (!filterstring.empty()) {
+                    if (!lcmatch(iter->second.gender_appropriate_name(u->male), filterstring)) {
+                        continue;
+                    }
+                }
+                sorted_scens.push_back(&(iter->second));
+            }
+            scens_length = sorted_scens.size();
+            if (scens_length == 0) {
+                popup(_("Nothing found.")); // another case of black box in tiles
+                filterstring.clear();
```

Comment:
> The filters are like in crafting? Couldn't they be like filters in advanced inventory (ie. displayed during edition)?

Mhm, sure could make a temp window on the right side for the help text... Don't think it's actually worth the effort and additional complexity - at least for now (I'd rather add actual "advanced filters" first). Now this brings a question of whether it would be a good idea to sync all filters to the same UI - methinks that yes :)